packages/ui/src/accounts/components/AccountItem/components/LocksDetails.tsx

```diff
@@ -6,28 +6,37 @@ import { Balances } from '@/accounts/types'
 import { RowGapBlock } from '@/common/components/page/PageContent'
 import { Label, TextMedium } from '@/common/components/typography'
 import { Address } from '@/common/types'
-import { useCurrentElection } from '@/council/hooks/useCurrentElection'
+import { useLatestElection } from '@/council/hooks/useLatestElection'
+import { useMyCastVotes } from '@/council/hooks/useMyCastVotes'
 
 interface LocksDetailsProps {
   balance: Balances | null
   address: Address
 }
 
 export const LocksDetails = ({ balance, address }: LocksDetailsProps) => {
-  const { election } = useCurrentElection()
+  const { election } = useLatestElection()
+  const { votes } = useMyCastVotes(election?.cycleId)
 
   const isActiveCandidate = useMemo(
     () => election?.candidates.find((candidate) => candidate.stakingAccount === address)?.status === 'ACTIVE',
     [election, address]
   )
 
+  const isVoteStakeLocked = useMemo(() => {
+    const candidate = votes?.find((vote) => vote.castBy === address)?.voteFor
+    // Disable recovering during election or if candidate won previous election
+    // Enable recovering if new council is elected
+    return !!(election || (candidate?.isCouncilMember && !election))
```

Comment:
Here `election` will be defined (unless there something wrong with the QN) but the question is whether:
1. A vote was cast in the latest election: the stake **shouldn't** be locked due to a vote cast several election ago
2. The latest election (where the vote was cast) is still going on, then the stake **should** be locked
3. The vote was cast to a winning candidate, then the stake **should** be locked

```suggestion
    return !!candidate && (!election?.isFinished || candidate.isCouncilMember)
```
(here if `election` is not defined the stake is not locked)

NB: AFAIK `isFinished` exist on the QN but not on the `Election` type so it should be added